src/main/java/org/sagebionetworks/web/client/widget/ReactComponent.java

```diff
@@ -50,37 +69,32 @@ public void run() {
     t.schedule(0);
   }
 
+  public void render(ReactElement<?, ?> reactElement) {
+    this.reactElement = reactElement;
+    createRootAndRender();
+  }
+
   @Override
   protected void onLoad() {
     super.onLoad();
     createRoot();
+
     if (reactElement != null) {
       this.render(reactElement);
     }
   }
 
   @Override
   protected void onUnload() {
-    if (root != null) {
-      // Asynchronously schedule unmounting the root to allow React to finish the current render cycle.
-      // https://github.com/facebook/react/issues/25675
-      Timer t = new Timer() {
-        @Override
-        public void run() {
-          root.unmount();
-          root = null;
-        }
-      };
-      t.schedule(0);
-    }
+    destroyRoot();
     super.onUnload();
   }
 
   @Override
   public void clear() {
-    // clear doesn't typically call onUnload, but we want to for this element.
-    this.onUnload();
-    super.clear();
+    if (root != null) {
+      root.render(React.createElement(React.Fragment));
+    }
```

Comment:
`onUnload` was inappropriate here. This widget can only contain a React component tree (no widget children), so it's much safer to just render a fragment. This prevents bugs where the root was unmounted or destroyed when it didn't need to be.